### ISSUE TITLE
Extend SC2045 to warn against x="$(ls file*glob.sh)"

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#1618]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2045
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/bin/bash
path=/a
x="$(ls "${path}/to/"versioned_file*.sh)"
cp "${x}" to/here/
call-me --edit "$(basename "${x}")"
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
A SC2045 (or SC2045-like) warning.
I don't have a solution off-hand, but it still looks like a mistake to do this
